src/core/layout/qgslayoutexporter.cpp

```diff
@@ -490,6 +490,8 @@ QgsLayoutExporter::ExportResult QgsLayoutExporter::exportToPdf( const QString &f
   ( void )contextRestorer;
   mLayout->renderContext().setDpi( settings.dpi );
 
+  mLayout->renderContext().setFlags( settings.flags );
```

Comment:
I'm not 100% sure, but this change may only have been ok in 3.6/master because of corresponding changes in the layout designer dialog. Can you look through the git blame on master for when I introduced this and please confirm that there's no related changes made in the designer to accompany this?

Reply:
@nyalldawson you're right an other commit is associated in the PR [needs-docs][layouts] Add checkbox to disable raster tiling for PDF/SVG exports #9016.

Do you think the other commit has to be backported ?
Does the Fix inconsistent use of layout render context flags  2752f83 introduced some regressions ?

Reply:
The issue fixed by your PR is https://issues.qgis.org/issues/19500 Layout export - raster divided into tiles, edges evident in pdf/svg

Reply:
@nyalldawson can I cherry-pick https://github.com/qgis/QGIS/commit/60b8d05278cd8133e1c06108ba875960d56e90e5#diff-b7c7a3b8898bc4177771ab86148a8348 ?

Reply:
This change is not linked to layout designer dialog. I have introduced flag in layout to print selection in server context. 
I have done the code for release 3.4.0 with the PR https://github.com/qgis/QGIS/pull/8320 [Server] Reactivate the capability to print selection with Server 3.4

Reply:
Hi @nyalldawson,

The `mLayout->renderContext().setFlags( settings.flags );` has been introduced before the **Refactor layout context** https://github.com/qgis/QGIS/commit/5bc543af6afaed1debe3471651ac1297694b7d70
https://github.com/qgis/QGIS/blame/master/src/core/layout/qgslayoutexporter.cpp#L356 in QGIS 3.0.0 in method `exportToImage` because this commit only rename the context access in layout.

The `mLayout->context().setFlags( settings.flags );` has been introduced before the **Work on PDF export ** https://github.com/qgis/QGIS/commit/91179f139669150c676492438d3e2edb11b7a3c7 in QGIS 3.0.0 in which you introduce `exportToPdf` and the `setFlags` missed.

In this commit you wrote
```  
  // If we are not printing as raster, temporarily disable advanced effects
  // as QPrinter does not support composition modes and can result
  // in items missing from the output
  mLayout->context().setFlag( QgsLayoutContext::FlagUseAdvancedEffects, settings.rasteriseWholeImage );
```
But flags are not set before.

The `mLayout->context().setFlags( settings.flags );` in `exportToImage` has been introduced by **Expose antialiasing option in image export dialog** https://github.com/qgis/QGIS/commit/2f0969e2bd5b5e74e6c18412d27f86a8f7c1072a#diff-7393cc3d739cdcd8a6531909637b63b5